# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/flyrank-bih/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

---

I chose **Lane 2 -> Refresh / Content Opportunity Scoring**.
- Companies have thousands of pages, and they can't manually review every page. So I want to build a system that helps prioritize which pages should be reviewed first, and This can save time and improve decision making.
---


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

---
1. The good research question is:

   > Which content pages should be reviewed first for refresh based on observable search and engagement signals?

2. The decision belongs to a **human**.

3. Action: The system simply tells the user to start with these pages.

4. Cost of a wrong recommendation:

   1. False Positive:

      > If it says "Review this page". But actually the page is fine -> The team wastes time.

   2. False Negative

      > The system ignores a page. But that page really needed updating -> The company may lose traffic.

---




## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [1]:
import os, sys, subprocess

IN_COLAB = "google.colab" in sys.modules
REPO_URL = "https://github.com/abuhussein1504/flyrank-ml-internship-starter"
REPO_DIR = "flyrank-ml-internship-starter"

if not os.path.isdir(REPO_DIR):
    subprocess.run(["git", "clone", "--depth", "1", REPO_URL, REPO_DIR], check=True)
os.chdir(REPO_DIR)
subprocess.run([sys.executable, "-m", "pip", "install", "-q", "-r", "requirements.txt"], check=True)

print("Working dir:", os.getcwd())
assert os.path.exists("data/raw/content_refresh_anonymized.csv"), "starter CSV not found — are you at the repo root?"
print("Starter data found. You're ready.")

Working dir: /content/flyrank-ml-internship-starter
Starter data found. You're ready.


In [17]:
import pandas as pd
df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

df.head().T

,0,1,2,3,4
content_id,content_304f48230142,content_a1fb4e703a9e,content_9aa793d4d895,content_331d6c4de07b,content_d99b7a2d90ca
client_id,client_f369cb89fc,client_4e07408562,client_7f2253d7e2,client_19581e27de,client_3fdba35f04
search_volume,10.0,90.0,0.0,10.0,0.0
competition,0.67,0.01,0.0,0.0,0.0
competition_level,HIGH,LOW,LOW,LOW,LOW
cpc,2.05,0.05,0.0,0.0,0.0
content_type,keyword article,keyword article,keyword article,keyword article,keyword article
main_intent,transactional,informational,informational,commercial,informational
word_count,3221.0,2481.0,3515.0,NaN,2803.0
char_count,20457.0,15562.0,23643.0,NaN,17469.0


In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 44 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   content_id              30000 non-null  object 
 1   client_id               30000 non-null  object 
 2   search_volume           27532 non-null  float64
 3   competition             27532 non-null  float64
 4   competition_level       27390 non-null  object 
 5   cpc                     27532 non-null  float64
 6   content_type            30000 non-null  object 
 7   main_intent             27626 non-null  object 
 8   word_count              22301 non-null  float64
 9   char_count              22301 non-null  float64
 10  provider_used           8562 non-null   object 
 11  model_used              24267 non-null  object 
 12  impressions_90d         30000 non-null  int64  
 13  clicks_90d              30000 non-null  int64  
 14  pageviews_90d           30000 non-null

In [16]:
print(df['trend_direction'].value_counts())

trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64


> The starter dataset contains 30,000 pages. Among them, 16,262 pages (54.2%) have a downward trend, while 5,962 pages (19.9%) are stable and 4,388 pages (14.6%) are growing. These numbers suggest that a large portion of the content may require attention, making page prioritization a meaningful problem to investigate.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

What my work can claim:

1. My work prioritizes pages for review.

2. The ranking provides decision support.

3. The model estimates which pages may deserve attention.

What my work can't claim:

1. My model proves that refreshing a page will increase traffic.

2. My model knows which pages Google likes.

3. My model guarantees higher rankings.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.